cmd/erasure-server-pool.go

```diff
@@ -739,7 +773,11 @@ func (z *erasureServerPools) DeleteObjects(ctx context.Context, bucket string, o
 	origIndexMap := map[int][]int{}
 	if !z.SinglePool() {
 		for j, obj := range objects {
-			idx, err := z.getPoolIdx(ctx, bucket, obj.ObjectName, 1<<30)
+			idx, err := z.getPoolIdxExisting(ctx, bucket, obj.ObjectName)
+			if err == errFileNotFound {
+				derrs[j] = toObjectErr(err)
+				continue
+			}
```

Comment:
This code is not needed since getPoolIdx will never return errFileNotFound @klauspost

Reply:
@harshavardhana `getPoolIdxExisting` will, though. Before a random index was sent and the `DeleteObjects` would have to fail for the error to be recorded. This avoid sending not found to DeleteObjects.